ENIGMAsystem/SHELL/Platforms/xlib/XLIBmain.cpp

```diff
@@ -194,14 +198,53 @@ void handleInput() {
 
 }  // namespace enigma
 
+namespace {
+
+int display_get_result = 0;
+
+int display_get_helper(unsigned i, int r) {
+  if (r != 0) return r; int result = 0, num_sizes; XRRScreenSize *xrrs;
+  Rotation original_rotation; Window root = DefaultRootWindow(enigma::x11::disp);
+  XRRScreenConfiguration *conf = XRRGetScreenInfo(enigma::x11::disp, root);
+  xrrs = XRRSizes(enigma::x11::disp, XDefaultScreen(enigma::x11::disp), &num_sizes);
+  SizeID original_size_id = XRRConfigCurrentConfiguration(conf, &original_rotation);
+  if (i == 2) result = XDisplayWidth(enigma::x11::disp, XDefaultScreen(enigma::x11::disp));
+  else if (i == 3) result = XDisplayHeight(enigma::x11::disp, XDefaultScreen(enigma::x11::disp));
+  
+  if (XineramaIsActive (enigma::x11::disp)) {
+    int m = 0; XineramaScreenInfo *xrrp = XineramaQueryScreens(enigma::x11::disp, &m);
+    if (i == 0) result = xrrp[original_size_id].x_org;
+    else if (i == 1) result = xrrp[original_size_id].y_org;
+    else if (i == 2) result = xrrs[original_size_id].width;
+    else if (i == 3) result = xrrs[original_size_id].height;
+    XFree(xrrp);
+  }
+  
+  r = result; return result;
+}
+
+} // anonymous namespace
+
 namespace enigma_user {
 
+int display_get_x() {
+  static int result = display_get_helper(0, display_get_result);
```

Comment:
Down here, you'd just say `display_get_helper().x_orig`, etc.

Reply:
therefore what you are asking is for a different way to accomplish the same thing while removing hardly 2 lines.

Reply:
I can do it but it's just a dumb excuse to wait for the ci all over again when this code does exactly fly what it is supposed to